node/src/components/block_executor.rs

```diff
@@ -343,15 +343,15 @@ impl BlockExecutor {
             // Read it from the storage.
             let height = finalized_block.height();
             effect_builder
-                .get_block_at_height_local(height - 1)
+                .get_block_at_height_from_storage(height - 1)
```

Comment:
We should never read directly from storage in my opinion – for accessing linear chain blocks we should be consulting `LinearChain` component.

Reply:
`LinearChain` will do this for you. Are you trying to cut on the reactor event count?